Documentation/admission-controller-usage.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Admission Controller
+weight: 2030
+indent: true
+---
+
+# Admission Controller
+
+An admission controller is a piece of code that intercepts requests to the Kubernetes API server prior to persistence of the object, but after the request is authenticated and authorized
+
+ There are two special controllers: MutatingAdmissionWebhook and ValidatingAdmissionWebhook.
+ 
+ Mutating controllers may modify the objects they admit but validation controllers are only allowed to validate requests.
+
+Currently, user can manipulate Custom Resource specs with any values which may result in Rook not functioning as expected. The present validation method in Kubernetes is the OpenAPI schema validation which can be used for basic validations like checking type of data, providing a range for the values etc but anything more complex (checking resource availability, network status, error handling) would not be possible under this scenario.
+
+An user with complex validation requirements for resources can benefit greatly from the use of admission controllers.
+
+## Quick Start 
+
+To deploy the Rook admission controllers we have a helper script that will automate the configuration.
+
+This script will help us achieve the following tasks
+1. Creates self-signed certificate.
+1. Creates a CSR for the certificate and gets it approved from the kubernetes cluster.
+1.  Stores these certificates as a secret.
+1. Creates a service account, clusterrole and clusterrolebindings for running the webhook server with minimal priviledges.
+1. Creates ValidatingWebhookConfig and fills the CA bundle with the appropriate value from the cluster.
+
+Run the following commands
+```
+kubectl create -f examples/kubernetes/ceph/common.yaml
+examples/kubernetes/ceph/admission-controller/deploy.sh
+```
+Now that the secrets have been deployed, we can deploy the operator and create a cluster using the following commands
+```
+kubectl create -f operator.yaml
+```
+
+At this point the operator will start the admission controller deployment automatically and the webhook will start intercepting requests for Rook resources. 
+
+## Certificate Management
+
+The script file does the job of creating a self signed kubernetes approved certificate and deploys it as a secret onto the cluster. It is mandatory that the secret is named "rook-admission-controller" as rook would try to detect a secret with such name before starting the admission controller servers. 
+
+The above approach of using self signed certificates is discouraged as it would be the job of the owner to maintain the certificates. The recommended approach would be to use a proper certificate manager and get signed certificates from a known Certificate authority. Once these are available, we have to create the secrets using the following command 
```

Comment:
```suggestion
The above approach of using self signed certificates is discouraged as it would be the job of the owner to maintain the certificates. The recommended approach would be to use a proper certificate manager and get signed certificates from a known Certificate authority. Once these are available, create the secrets using the following command 
```